internal/powerplatform/services/data_record/api_data_record.go

```diff
@@ -0,0 +1,432 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package powerplatform
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	constants "github.com/microsoft/terraform-provider-power-platform/constants"
+	api "github.com/microsoft/terraform-provider-power-platform/internal/powerplatform/api"
+	powerplatform_helpers "github.com/microsoft/terraform-provider-power-platform/internal/powerplatform/helpers"
+)
+
+func NewDataRecordClient(api *api.ApiClient) DataRecordClient {
+	return DataRecordClient{
+		Api: api,
+	}
+}
+
+type DataRecordClient struct {
+	Api *api.ApiClient
+}
+
+type EnvironmentIdDto struct {
+	Id         string                     `json:"id"`
+	Name       string                     `json:"name"`
+	Properties EnvironmentIdPropertiesDto `json:"properties"`
+}
+
+type EnvironmentIdPropertiesDto struct {
+	LinkedEnvironmentMetadata LinkedEnvironmentIdMetadataDto `json:"linkedEnvironmentMetadata"`
+}
+
+type LinkedEnvironmentIdMetadataDto struct {
+	InstanceURL string
+}
+
+type EntityDefinitionsDto struct {
+	OdataContext          string `json:"@odata.context"`
+	PrimaryIDAttribute    string `json:"PrimaryIdAttribute"`
+	LogicalCollectionName string `json:"LogicalCollectionName"`
+	MetadataID            string `json:"MetadataId"`
+}
+
+type RelationApiResponse struct {
+	OdataContext string            `json:"@odata.context"`
+	Value        []RelationApiBody `json:"value"`
+}
+
+type RelationApiBody struct {
+	OdataID string `json:"@odata.id"`
+}
+
+func getEntityDefinition(ctx context.Context, client *DataRecordClient, environmentUrl string, entityLogicalName string) *EntityDefinitionsDto {
+	e, _ := url.Parse(environmentUrl)
+	entityDefinitionApiUrl := &url.URL{
+		Scheme:   e.Scheme,
+		Host:     e.Host,
+		Path:     fmt.Sprintf("/api/data/%s/EntityDefinitions(LogicalName='%s')", constants.DATAVERSE_API_VERSION, entityLogicalName),
+		Fragment: "$select=PrimaryIdAttribute,LogicalCollectionName",
+	}
+	entityDefinition := EntityDefinitionsDto{}
+	_, err := client.Api.Execute(ctx, "GET", entityDefinitionApiUrl.String(), nil, nil, []int{http.StatusOK}, &entityDefinition)
+	if err != nil {
+		return nil
```

Comment:
api methods should return errors, not hide them

Reply:
done, thanks 👍